sdk/messaging/azeventhubs/checkpoints/blob_store.go

```diff
@@ -36,9 +37,16 @@ type BlobStoreOptions struct {
 // NewBlobStore creates a checkpoint store that stores ownership and checkpoints in
 // Azure Blob storage.
 // NOTE: the container must exist before the checkpoint store can be used.
-func NewBlobStore(containerClient *container.Client, options *BlobStoreOptions) (*BlobStore, error) {
-	return &BlobStore{
+func NewBlobStore(containerClient *container.Client, options *BlobStoreOptions) (*azeventhubs.CheckpointStore, error) {
```

Comment:
Does this have to return a `CheckpointStore` type?  I thought that abstraction was only for inputs.

Reply:
It has to since we need to pass the result of this into the Processor() when it's created (and that type is of `CheckpointStore`.

Reply:
I see.  It just seems weird that the name of the func is `NewBlobStore()` but it returns something other than a `BlobStore` (I get why).

Does this mean then that the `BlobStore` type doesn't need to be exported?

Reply:
You're very correct, I will unexport the `BlobStore` type.